fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanScheduleValidator.java

```diff
@@ -176,4 +178,24 @@ private void validateRepaymentsStartingFromDateIsAfterDisbursementDate(final Lis
             dataValidationErrors.add(error);
         }
     }
+
+    public boolean validateDownPaymentAttribute(final boolean isDownPaymentEnabledInLoanProduct, final JsonElement element) {
+        boolean isDownPaymentEnabled = isDownPaymentEnabledInLoanProduct;
```

Comment:
This method is overcomplicated:
```
public void validateDownPaymentAttribute(final boolean isDownPaymentEnabledInLoanProduct, final JsonElement element) {
      Boolean inputIsDownPaymentEnabled = this.fromApiJsonHelper.extractBooleanNamed(LoanProductConstants.ENABLE_DOWN_PAYMENT, element);
      if(!isDownPaymentEnabledInLoanProduct && Boolean.TRUE.equals(inputIsDownPaymentEnabled)) {
               throw new GeneralPlatformDomainRuleException("error.msg.downpayment.is.not.enabled.in.loan.product",
                        "The Loan can not override the downpayment flag because in the Loan Product the downpayment is disabled",
                        inputIsDownPaymentEnabled);
                        }
}
```

What do you think?